### PR TITLE
Fix -march=native once and for all (?)

### PR DESCRIPTION
**Pull request by mrks**
This should enable `-march=native` for exactly those architecture/compiler combinations that support it:

https://github.com/hyrise/hyrise/pull/2306#issuecomment-767388729

- [x]  M1: builds with clang and no performance regression
- [x]  Raspberry PI: builds with clang/gcc and no performance regressions

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -79,15 +79,13 @@ set(FLAGS_ALL "-fopenmp-simd")  # enables loop vectorization hints, but does not
...
-# https://community.arm.com/developer/tools-software/tools/b/tools-software-ides-blog/posts/compiler-flags-across-architectures-march-mtune-and-mcpu
-# We check specifically for arm64 (e.g., Apple's M1) or aarch64 (e.g., Raspberry Pi 4), as these are the two ARM architectures we have tested so far.
-# Match for "arm" on your own risk to include other ARM architures.
-if (CMAKE_SYSTEM_PROCESSOR MATCHES "arm64|aarch64")
-    set(FLAGS_RELEASE "-mcpu=native")
-else()
-    set(FLAGS_RELEASE "-march=native")  
+# Build the binary optimized for the current system, ignoring older systems.
+# Not all environments support march=native - check before we use it.
+set(FLAGS_RELEASE "-mcpu=native -mtune=native")
```
Comment (Bouncner) [inline CMakeLists.txt:84]: The reason for not setting all three was the following summary of the linked article (it's an ARM article):
```
TL;DR: Whenever possible use only -mcpu=native. Avoid -march and -mtune.
```

**Comment by Bouncner** [comment]
Works with Clang and GCC on a recent Raspberry Pi with Ubuntu 20.04 (did not wait for the full compilation, takes quite some time).
On the M1, cmake runs for both compilers, but GCC does not compile (probably expected).

File: CMakeLists.txt (hunk 1 above)
Comment (mrks) [reply CMakeLists.txt:84]: My understand is that this summary is limited to ARM. As far as I know, march is the way to go for x86. That contradicts the tldr.

Comment (Bouncner) [reply CMakeLists.txt:84]: Hmm, what you said has been done before this PR. Why does this contradict the tldr?

Comment (mrks) [reply CMakeLists.txt:84]: Are you talking about the addition of mtune? As far as I understand it, it doesn't hurt, but `mcpu` is more specific in some cases.

**Comment by mrks** [comment]
Hmpf.

> clang: warning: argument unused during compilation: '-mcpu=native' [-Wunused-command-line-argument]

**Comment by mrks** [comment]
Removed mtune, fixed clang unused warning.

**Review by Bouncner — APPROVED** [review]

**Comment by Bouncner** [comment]
Zünden?

**Comment by mrks** [comment]
Did you re-run it on the machines? I'd rather double check this, possibly even the performance, as any error that we introduce here will likely not be caught any time soon. I'm also hoping on feedback from @pkoutsov.

**Comment by Bouncner** [comment]
> Did you re-run it on the machines? I'd rather double check this, possibly even the performance, as any error that we introduce here will likely not be caught any time soon. I'm also hoping on feedback from @pkoutsov.

M1 works fine. There has been a 9% improvement on Q9. When I ran it again it was 6% but then  Q22 improved by 7%. So I would assume is just flaky with Q9 right now on M1.

The RaspPi killed the compiler because of a lack of swap space. Compiling with `-j 1` now, that's going to take a while.

**Comment by Bouncner** [comment]
Everything's fine on a Raspberry Pi 4.

**Comment by pkoutsov** [comment]
Everything's fine on ```ppc64le``` as well
